controller/dataplane/owned_deployment.go

```diff
@@ -306,17 +306,64 @@ func reconcileDataPlaneDeployment(
 
 		k8sresources.SetDefaultsPodTemplateSpec(&desired.Spec.Template)
 
+		// Store the original replica count before updating metadata
+		originalReplicas := existing.Spec.Replicas
+
 		// ensure that object metadata is up to date
 		updated, existing.ObjectMeta = k8sutils.EnsureObjectMetaIsUpdated(existing.ObjectMeta, desired.ObjectMeta)
 
+		// Check if this is a deployment restart (kubectl rollout restart)
+		// We need to check if the existing deployment has the restart annotation
+		// This is different from our previous approach because the desired deployment
+		// is generated from the DataPlane resource and won't have the restart annotation
+		isRestartOperation := false
+		if existing.Spec.Template.Annotations != nil {
+			_, hasRestartAnnotation := existing.Spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"]
+			if hasRestartAnnotation {
+				isRestartOperation = true
+				log.Debug(logger, "detected restart operation", "isRestartOperation", isRestartOperation)
+			}
+		}
```

Comment:
Can we extract this into a `func` like: `PodTemplateSpecHasRestartAnnotation()` which could return the annotation value and a `boolean` indicating if it was found?

Reply:
Yes, addressed.